frontend/dockerfile/linter/ruleset.go

```diff
@@ -88,13 +88,6 @@ var (
 			return fmt.Sprintf("Relative workdir %q can have unexpected results if the base image changes", workdir)
 		},
 	}
-	RuleUndefinedArg = LinterRule[func(string) string]{
-		Name:        "UndefinedArg",
-		Description: "ARGs should be defined before their use",
-		Format: func(arg string) string {
-			return fmt.Sprintf("Usage of undefined variable '$%s'", arg)
-		},
-	}
```

Comment:
@dvdksn Related to your comment https://github.com/crazy-max/buildkit/pull/18#discussion_r1627630984, I guess we could just comment this part in the meantime if not used.

cc @colinhemmings @daghack @tonistiigi